### DISCUSSION TITLE
API Route giving 404 in production, working locally

### DISCUSSION THREAD

**Discussion by brokerxing** [vercel/community#4100] [Help]
### Summary

I am simply trying to get auth0 setup - I've done this over a dozen times with no issue.  Using latest Next (13) I have this structure:

pages/api/auth/[...auth0].js

I run it locally and it works perfectly.

I deploy to Vercel and I get 404.  In fact, I can't seem to be able to ANY routes to work at all when in production.

This give 404 too...but works locally just fine:

pages/api/test

export default function handler(req, res) {
    res.status(200).json({ name: 'John Doe' })
  }

Any ideas?  

### Example

https://exchange-ochre.vercel.app/api/test

### Steps to Reproduce

I don't know how to reproduce in prod.

**Comment by brokerxing** [comment]
![routeissue](https://github.com/vercel/community/assets/140661708/cf29249b-df87-49cb-b245-ec1e8a54edeb)

why is this 404 in production???

**Reply by javierfuentesm** [reply]
@brokerxing did you solve it ? im getting this same error works locally and also in production, but it fails when I use the API through my bot it doesn't make sense in the vercel logs it says 404 but I click the link and I get a response even in production

**Reply by adomakins** [reply]
@javierfuentesm did you solve it bro lmao same thing here

**Reply by javierfuentesm** [reply]
@adomakins  here you have https://github.com/orgs/vercel/discussions/5619#discussioncomment-8338626

**Reply by adomakins** [reply]
@javierfuentesm Damn thanks for the response bro! I ended up fixing it another way lol

**Reply by adomakins** [reply]
Here's my issue and solution for future sufferers https://github.com/orgs/vercel/discussions/7439#discussioncomment-10027435
